### ISSUE TITLE
Cannot add Kubernetes resource

### ISSUE THREAD

**Issue by croemmich** [microsoft/azure-devops-go-api#116]
https://github.com/microsoft/azure-devops-go-api/blob/00dac5c867394a3c5ca4e12b6965d7625a1588c6/azuredevops/v6/taskagent/client.go#L352

 provides no means of supplying a `serviceEndpointId`. This makes it impossible to use with out receiving a 500 from the server with:
```
{
    "$id": "1",
    "innerException": null,
    "message": "Failed to create resource. Please ensure that all mandatory fields are provided as input.",
    "typeName": "Microsoft.TeamFoundation.DistributedTask.WebApi.KubernetesResourceException, Microsoft.TeamFoundation.DistributedTask.WebApi",
    "typeKey": "KubernetesResourceException",
    "errorCode": 0,
    "eventId": 3000
}
```

<br/>
<br/>

The  `AddKubernetesResourceArgs`:
https://github.com/microsoft/azure-devops-go-api/blob/00dac5c867394a3c5ca4e12b6965d7625a1588c6/azuredevops/v6/taskagent/client.go#L384

should probably take a:
https://github.com/microsoft/azure-devops-go-api/blob/00dac5c867394a3c5ca4e12b6965d7625a1588c6/azuredevops/v6/taskagent/models.go#L964

instead of the current:
https://github.com/microsoft/azure-devops-go-api/blob/00dac5c867394a3c5ca4e12b6965d7625a1588c6/azuredevops/v6/taskagent/models.go#L956

**Comment by xuzhang3** [comment]
@nechvatalp    Can you  fix this SDK bug? `serviceendpointId` is a  required parameter to  create environment kubenetes resources and cannot be ignored. For API `AddKubernetesResource` , the args should use `KubernetesResourceCreateParametersExistingEndpoint` instead of `KubernetesResourceCreateParameters`

**Comment by theitalianz** [comment]
@croemmich / @xuzhang3  I get the same 500 if `serviceEndpointId` is omitted but where did you find it in the documentation?
I'm looking at https://docs.microsoft.com/en-us/rest/api/azure/devops/distributedtask/kubernetes/add?view=azure-devops-rest-6.0
but it is not mentioned in the request body 🤔

**Comment by xuzhang3** [comment]
@theitalian I get the request body by browser dev tools.

**Comment by theitalianz** [comment]
@xuzhang3 sneaky! Ok, so docs needs to be update too seems like. Gonna create an issue there as well.

**Comment by xuzhang3** [comment]
@theitalianz I didn't test all the ways to create it, from the swagger doc there should be 3 ways to create a new environment.

**Comment by croemmich** [comment]
Thanks for taking a look at this @theitalianz and @xuzhang3!

**Comment by kexugit** [comment]
@croemmich @theitalianz @xuzhang3 Please make sure swagger files are also updates - reference documents published on Docs will reflect what's defined in the swagger files.

**Comment by nechvatalp** [comment]
@croemmich Thanks for reporting, it actually should accept `KubernetesResourceCreateParametersNewEndpoint` the API accepts `KubernetesResourceCreateParameters` which is the baseclass of `KubernetesResourceCreateParametersNewEndpoint`. 
However this causes generated documentation and clients to miss the crucial parameter, I filed a bug on the team owning this API to fix it.

**Comment by croemmich** [comment]
@nechvatalp I don't think `KubernetesResourceCreateParametersNewEndpoint` will work for my use case. I don't want to create the K8s endpoint. It's already created, I just want to associate it.

This DOES NOT work:
```
POST: https://dev.azure.com/{{organization}}/{{project}}/_apis/distributedtask/environments/{{envId}}/providers/kubernetes?api-version=7.1-preview.1

{"name":"test","namespace":"test","endpoint":{"id":"615631d7-2658-481f-b68e-84b6a7a2c23a"}}
```
Returns:
```
{
    "$id": "1",
    "innerException": null,
    "message": "Specified method is not supported.",
    "typeName": "System.NotSupportedException, mscorlib",
    "typeKey": "NotSupportedException",
    "errorCode": 0,
    "eventId": 0
}
```

This is the request body I need to make:
```
POST: https://dev.azure.com/{{organization}}/{{project}}/_apis/distributedtask/environments/{{envId}}/providers/kubernetes?api-version=7.1-preview.1

{"name":"test","namespace":"test","serviceEndpointId":"615631d7-2658-481f-b68e-84b6a7a2c23a"}
```

**Comment by nechvatalp** [comment]
Sorry, unfortunately I don't have enough time to properly maintain this repository. It comes down to priorities I have other work which is considered higher priority. I am trying to push on our PMs and management to invest more into our API clients as I would personally love to have them properly taken care of but so far, I was unsuccessful.

**Comment by nechvatalp** [comment]
This should be fixed in the latest 7.1 release.
